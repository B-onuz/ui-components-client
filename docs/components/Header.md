# Header (Component)

Componente de cabeçalho.

## Como utilizar

```js
import Header from 'ui-components/lib/components/Header'
```

```jsx
<Header title="Texto do título" subTitle="Subtitulo da página" />
```

## Styled-system props

- typography https://styled-system.com/table#typography
