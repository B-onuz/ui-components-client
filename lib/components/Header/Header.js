"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Title = _interopRequireDefault(require("./Title"));

var _SubTitle = _interopRequireDefault(require("./SubTitle"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Header/Header.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  flex: 1;\n  font-family: inherit;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledHeader = _styledComponents["default"].header(_templateObject(), _styledSystem.typography);

var Header = function Header(_ref) {
  var children = _ref.children,
      title = _ref.title,
      subTitle = _ref.subTitle,
      _ref$textAlign = _ref.textAlign,
      textAlign = _ref$textAlign === void 0 ? 'left' : _ref$textAlign,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "title", "subTitle", "textAlign"]);

  return /*#__PURE__*/_react["default"].createElement(StyledHeader, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 5
    }
  }), !!title && /*#__PURE__*/_react["default"].createElement(_Title["default"], {
    textAlign: textAlign,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 16,
      columnNumber: 19
    }
  }, title), !!subTitle && /*#__PURE__*/_react["default"].createElement(_SubTitle["default"], {
    textAlign: textAlign,
    mt: 1,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 9
    }
  }, subTitle));
};

Header.defaultProps = {
  fontSize: [2, 3, 4]
};
var _default = Header;
exports["default"] = _default;
module.exports = exports.default;