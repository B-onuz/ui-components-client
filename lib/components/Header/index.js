"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Header = _interopRequireDefault(require("./Header"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Header["default"];
exports["default"] = _default;
module.exports = exports.default;