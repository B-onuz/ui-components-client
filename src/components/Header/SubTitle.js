import styled from 'styled-components'
import { color, space, typography, compose } from 'styled-system'

const SubTitle = styled.h1`
  font-size: 16px;
  color: #00000099;
  font-weight: 400;
  margin: 0;
  ${compose(color, space, typography)}
`

export default SubTitle
