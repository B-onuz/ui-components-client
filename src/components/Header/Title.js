import styled from 'styled-components'
import { color, space, typography, compose } from 'styled-system'

const Title = styled.h1`
  margin: 0;
  ${compose(color, typography, space)}
  font-size: 1em !important;
  @media only screen and (min-width: 468px) {
    white-space: nowrap;
    margin-left: 8px;
  }
`

Title.defaultProps = {
  color: 'darkRed',
}

export default Title
