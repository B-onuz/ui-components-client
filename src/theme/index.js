const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
