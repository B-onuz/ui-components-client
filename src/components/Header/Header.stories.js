import React from 'react'
import Header from './Header'

export default {
  title: 'Header',
  component: Header,
}

export const HeaderExample = () => <Header title={'Texto do título'} subTitle={'Subtitulo da página'} />
