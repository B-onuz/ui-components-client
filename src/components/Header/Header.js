import React from 'react'
import styled from 'styled-components'
import { typography } from 'styled-system'
import Title from './Title'
import SubTitle from './SubTitle'

const StyledHeader = styled.header`
  ${typography}
  flex: 1;
  font-family: inherit;
`

const Header = ({ children, title, subTitle, textAlign = 'left', ...rest }) => {
  return (
    <StyledHeader {...rest}>
      {!!title && <Title textAlign={textAlign}>{title}</Title>}
      {!!subTitle && (
        <SubTitle textAlign={textAlign} mt={1}>
          {subTitle}
        </SubTitle>
      )}
    </StyledHeader>
  )
}

Header.defaultProps = {
  fontSize: [2, 3, 4],
}

export default Header
