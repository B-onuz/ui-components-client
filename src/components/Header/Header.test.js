import React from 'react'
import Header from './Header.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Header changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Header />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
