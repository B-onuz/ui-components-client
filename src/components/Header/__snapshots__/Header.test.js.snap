// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header changes the class when hovered 1`] = `
.c0 {
  font-size: 16px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: inherit;
}

@media screen and (min-width:40em) {
  .c0 {
    font-size: 20px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    font-size: 24px;
  }
}

<header
  className="c0"
  fontSize={
    Array [
      2,
      3,
      4,
    ]
  }
/>
`;
